### ISSUE TITLE
hidewhens are confusing

### ISSUE THREAD

**Issue by djay** [plomino/Plomino#631]
# User problem

I've noticed several common errors with hidewhens that make it hard to use
1. Mostly people I saw users have something hidden by default and have it appear when a certain option is picked. They are thinking more in terms of a "showwhen"
2. The conditions for checkboxes and radio buttons are different. Checkboxes are lists and radios are single values and due to no type checking hidewhen will just get stuck as true or false.

# options
1. The formula helper idea in https://github.com/plomino/Plomino/issues/618 would allow for a helper for common cases like show when field X, option Y is picked. This should be out of the box I think
2. change hidewhens to showwhens.
  - con: backwards compatibility.
3. introduce not: into hidewhen syntax and UI. This would let you create hidewhens that match the showwhen logic. e.g. hidewhen start:not:over_18.
  - also means you only have to create a single hidewhen for a yes/no case which is nice.
4. Some kind of way to debug? For instance a debug mode that dynamically shows all current picked values as hidewhen type formulas so you can just cut and paste?
5. ?

**Comment by ebrehault** [comment]
You are right, the hidewhen principle is really twisted. It actually comes directly from Lotus Notes, and when I have started Plomino, I tried to keep as close as possible to Lotus Notes principles and naming (like Agent, Views/Columns, etc.).

+1 on renaming into showwhen and +1 on managing the `not:` directive.

**Comment by djay** [comment]
> On 22 Jun 2015, at 3:37 pm, Eric BREHAULT <notifications@github.com> wrote:
> 
> You are right, the hidewhen principle is really twisted. It actually comes directly from Lotus Notes, and when I have started Plomino, I tried to keep as close as possible to Lotus Notes principles and naming (like Agent, Views/Columns, etc.).
> 
> +1 on renaming into showwhen and +1 on managing the not: directive.

I was thinking you can’t rename it to showwhen because of incompatiblity reasons.
But I guess if you introduce not: at the same time you could just render a hidewhen as a not:showwhen.

**Comment by djay** [comment]
some nice debug output here - https://wp-types.com/documentation/user-guides/conditional-html-output-in-views/
